processing/src/test/java/org/apache/druid/query/scan/ScanResultValueFramesIterableTest.java

```diff
@@ -222,6 +262,71 @@ public void testBatchingWithHeterogenousAndEmptyScanResultValues()
     );
   }
 
+  @Test
+  public void testBatchingWithHeterogenousAndEmptyScanResultValuesAndNullTypes()
+  {
+    List<FrameSignaturePair> frames = Lists.newArrayList(
+        createIterable(
+            scanResultValue1(0),
+            scanResultValue2(0),
+            scanResultValue1(2),
+            scanResultValue1(0),
+            scanResultValue2(2),
+            scanResultValue2(0),
+            scanResultValue2(0)
+        )
+    );
+    Assert.assertEquals(2, frames.size());
+    QueryToolChestTestHelper.assertArrayResultsEquals(
+        ImmutableList.of(
+            new Object[]{1L, 1.0D},
+            new Object[]{2L, 2.0D}
+        ),
+        new FrameBasedInlineDataSource(Collections.singletonList(frames.get(0)), SIGNATURE1).getRowsAsSequence()
+    );
+    QueryToolChestTestHelper.assertArrayResultsEquals(
+        ImmutableList.of(
+            new Object[]{3.0D, 3L},
+            new Object[]{4.0D, 4L}
+        ),
+        new FrameBasedInlineDataSource(Collections.singletonList(frames.get(1)), SIGNATURE2).getRowsAsSequence()
+    );
+  }
+
+  @Test
+  public void testBatchingWithDifferentRowSignaturesButSameTrimmedRowSignature()
+  {
+    List<FrameSignaturePair> frames = Lists.newArrayList(
+        createIterable(
+            scanResultValue3(0),
+            scanResultValue4(0),
+            scanResultValue3(2),
+            scanResultValue3(0),
+            scanResultValue4(2),
+            scanResultValue4(0),
+            scanResultValue3(0)
+        )
+    );
+    Assert.assertEquals(1, frames.size());
+    QueryToolChestTestHelper.assertArrayResultsEquals(
+        ImmutableList.of(
+            new Object[]{5.0D, 5L},
+            new Object[]{6.0D, 6L},
+            new Object[]{7.0D, 7L},
+            new Object[]{8.0D, 8L}
+        ),
+        new FrameBasedInlineDataSource(Collections.singletonList(frames.get(0)), SIGNATURE2).getRowsAsSequence()
```

Comment:
its very odd to see `Assert.assertEquals(1, frames.size())` and then after a few lines later `Collections.singletonList(frames.get(0))` - why not just `frames` ?